spec/format_parser_spec.rb

```diff
@@ -189,12 +189,26 @@
         'FormatParser::CR3Parser',
         'FormatParser::DPXParser',
         'FormatParser::FLACParser',
-        'FormatParser::MP3Parser',
         'FormatParser::OggParser',
         'FormatParser::TIFFParser',
-        'FormatParser::WAVParser'
+        'FormatParser::WAVParser',
+        'FormatParser::MP3Parser'
       ])
     end
+
+    it 'ensures that MP3 parser is the last one among all' do
+      parsers = FormatParser.parsers_for(
+        [:audio, :image, :document, :text, :video, :archive],
+        [
+          :aac, :aiff, :arw, :bmp, :cr2, :cr3, :dpx, :fdx, :flac, :gif, :heif, :heic,
```

Comment:
_(Same comment as that from `spec/remote_fetching_spec.rb`)_ 

> It feels more future-proof to get this programatically, maybe exposing FormatParser's `@parsers_per_nature` somehow? or exposing all parsers as an array in proper execution order?

Reply:
Should we expose an internal state just because it would simplify some tests? 🤔

Reply:
There are many other places in this spec file that list specific natures and formats.

Reply:
@linkyndy I don't see an issue in FormatParser exposing a list of supported formats -- or natures. The parsers themselves can be isolated as internals, but the full support list can be interesting to consumers.

Reply:
@CostinTanasoiu the other ones I see as less of an issue because they are fine testing a subset. This one is tricky because it really wouldn't do it's job if someone creates a new parser in the future with a lower priority than mp3-- it only covers updates.

Reply:
@rcpalacio I've rewritten the test. I exposed two new variables from FormatParser –`registered_natures` and `registered_formats` – and I'm using these to retrieve the parsers in my test. Let me know if this isn't ideal or if there's a better approach.

Reply:
I think it looks good 👍